pensieve/external_config.py

```diff
@@ -0,0 +1,91 @@
+"""
+Retrieves external configuration files for specific experiments.
+
+Experiment-specific configuration files are stored in https://github.com/mozilla/pensieve-config/
+"""
+
+import datetime as dt
+import attr
+from dateutil import parser
+from github import Github
+from github.ContentFile import ContentFile
+from google.cloud import bigquery
+import os
+import re
+import toml
+from typing import List, Optional
+
+from pensieve.config import AnalysisSpec
+
+
+@attr.s(auto_attribs=True)
+class ExternalConfig:
+    """Represent an external config file."""
+
+    normandy_slug: str
+    spec: AnalysisSpec
+    last_modified: dt.datetime
+
+    def updated(self, bq_project: str, bq_dataset: str) -> bool:
+        """
+        Check whether the config file has been updated/added and
+        associated BigQuery tables are out of date.
+        """
```

Comment:
it seems like we'll have to invoke this query maybe dozens of times as we start accumulating config files in the repository; what if we had a function that accepted an ExternalConfigCollection and a dataset and returned a list of which experiments had stale data?

this seems like something that might ergonomic with the REST API vs the query API